drivers/flash/flash_shell.c

```diff
@@ -732,6 +732,19 @@ static int cmd_page_info(const struct shell *sh, size_t argc, char *argv[])
 	return 0;
 }
 
+#define PRINT_PARTITION_INFO(part) shell_print(sh, "%-15s 0x%08x %d Kb", DT_PROP(part, label), \
+					  DT_REG_ADDR(part), DT_REG_SIZE(part) / 1024);
+
+static int cmd_partitions(const struct shell *sh, size_t argc, char *argv[])
+{
+#if DT_NODE_HAS_STATUS_OKAY(DT_CHOSEN(zephyr_flash_controller))
+	DT_FOREACH_CHILD(DT_CHILD(DT_CHOSEN(zephyr_flash), partitions), PRINT_PARTITION_INFO);
```

Comment:
Partitions can be declared outside of the chosen `zephyr,flash` node, why don't you iterate `fixed-partitions` compatibles?

Reply:
good idea, thx @pdgendt